### ISSUE TITLE
Webpack Configuration & Source Map

### ISSUE THREAD

**Issue by eileenchoe** [dondi/GRNsight#577]
When merging betaa (which now includes webpack) into my node coloring branch, I discovered that since webpack bundles and minifies all the web-client code, it is now very difficult to debug. I think that source mapping might be the solution to this: https://www.html5rocks.com/en/tutorials/developertools/sourcemaps/
@dondi have you come across this before?

Additionally, I added `webpack --watch` to the parallelshell list in `npm start`. 

Lastly, I had to do `npm install -g webpack` and `npm install -g webpack-cli` to get webpack working on my machine. We'll either have to add this to the development workflow or find out a way to automate it.

**Comment by dondi** [comment]
@eileenchoe Yes, generating source map files is indeed how you can see readable code when working with a minified codebase. The other possibility is to turn off minification when running webpack locally, minifying only when deploying for release. This would be a new model for release though and may require some other scripting changes, so in the end, having a source map file handy is the way to go.

`webpack --watch` should likely be standard now, yes, but finally I have generally been OK doing a local install of `webpack` (i.e., not `-g`). Was there a particular reason that you felt `-g` was necessary? If it is a matter of paths (i.e., finding executables locally), then having the entry `./node_modules/.bin` in your `PATH` should do the trick.

**Comment by eileenchoe** [comment]
Got it. I'm taking a look at Webpack's documentation for prod/dev setup and will try implementing this weekend: https://webpack.js.org/guides/production/

I was getting errors before installing webpack globally, but I did not have `./node_modules/.bin` in my PATH. Deleting webpack and webpack-cli from global node modules folder, then adding node_modules in my path, and running webpack sucessfully confirmed that the global installation is not necessary.

**Comment by eileenchoe** [comment]
Ah, the simplest solution is just passing in production or development as an argument to webpack-cli. Production will minify, and development will retain source mapping. 

I've split out the start up script to handle these two situations, so `npm run start-dev` will start the server in dev mode and `npm run start-prod` in prod mode. @dondi @yshin4 what do you think of this?

```
    "start-dev": "parallelshell --wait \"nodemon web-client/app.js\" \"nodemon server/app.js\" \"webpack --watch --mode=development\"",
    "start-prod": "parallelshell --wait \"nodemon web-client/app.js\" \"nodemon server/app.js\" \"webpack --watch --mode=production\"",
```

**Comment by dondi** [comment]
Noted in the meeting by @dondi is that typically the distinction between `dev` and `prod` is conveyed via environment variable. But so as not to derail the central work, the suggestion is to timebox this and if it can't be adjusted quickly enough then we can move to it later on.

**Comment by eileenchoe** [comment]
Decided to move forward with the two start scripts for dev and prod. This work will be merged with the PR for #566

**Comment by eileenchoe** [comment]
Live on beta v.2.3.6. Npm installed on the GRNsight server, so correct versions of webpack were installed. Additionally changed, the script to `npm run start-prod` and verified that webpack is minifying output for the production code.

**Comment by eileenchoe** [comment]
Updated documentation with the new start scripts

**Comment by kdahlquist** [comment]
I'm leaving this one for @dondi to review and close.
